### PR TITLE
[Profiler] Add Optional Flag to turn off external correlations

### PR DESCRIPTION
**Pull request by sraikund16**
Summary: External Correlations are super spammy and oftentimes not even useful. Add flag during init to remove them entirely

Test Plan: https://www.internalfb.com/intern/perfdoctor/trace_view?filepath=tree/traces/dynocli/devvm2185.cco0.facebook.com/rank-0.Dec_10_12_33_31.531106.pt.trace.json.gz&bucket=gpu_traces

Differential Revision: D67048206

cc @robieta @chaekit @guotuofeng @guyang3532 @dzhulgakov @davidberard98 @briancoutinho @sanrise

### CODE REVIEW

File: test/profiler/test_profiler.py [hunk 1]
```diff
@@ -2122,6 +2122,55 @@ def test_skip_first_wait(self):
...
+        # Run with External Id for CUDA events on
+        disable_external_correlation = False
+        with profile(
+            activities=activities,
+            experimental_config=torch._C._profiler._ExperimentalConfig(
+                disable_external_correlation=disable_external_correlation
+            ),
+        ) as prof_on:
+            self.payload(use_cuda=True)
+        validate_json(prof_on, disable_external_correlation)
```
Comment (ngimel) [inline test/profiler/test_profiler.py:2161]:
```suggestion
        for disable_external_correlation in (True, False):
            with profile(
                activities=activities,
                experimental_config=torch._C._profiler._ExperimentalConfig(
                     disable_external_correlation=disable_external_correlation
                ),
            ) as prof:
                self.payload(use_cuda=True)
            validate_json(prof disable_external_correlation)
```

**Review by ngimel — APPROVED** [review]

**Comment by huydhn** [comment]
@pytorchbot revert -m 'Sorry for reverting your change but the test is failing on ROCm' -c nosignal

Due to https://github.com/pytorch/pytorch/issues/143233, the failures from ROCm were hidden on the PR.  As the sev has been resolved, please help fix and reland this change

**Pull request closed without merging**